frost-core/src/frost/keys/dkg.rs

```diff
@@ -331,6 +328,56 @@ where
     Some(Challenge(C::HDKG(&preimage[..])?))
 }
 
+/// Compute the proof of knowledge of the secret coefficients used to generate
+/// the public secret sharing commitment.
+#[cfg_attr(feature = "internals", visibility::make(pub))]
+pub(crate) fn compute_proof_of_knowledge<C: Ciphersuite, R: RngCore + CryptoRng>(
+    identifier: Identifier<C>,
+    coefficients: &[Scalar<C>],
+    commitment: &VerifiableSecretSharingCommitment<C>,
+    mut rng: R,
+) -> Result<Signature<C>, Error<C>> {
+    // Round 1, Step 2
+    //
+    // > Every P_i computes a proof of knowledge to the corresponding secret
+    // > a_{i0} by calculating σ_i = (R_i, μ_i), such that k ← Z_q, R_i = g^k,
+    // > c_i = H(i, Φ, g^{a_{i0}} , R_i), μ_i = k + a_{i0} · c_i, with Φ being
+    // > a context string to prevent replay attacks.
+    let k = <<C::Group as Group>::Field>::random(&mut rng);
+    let R_i = <C::Group>::generator() * k;
+    let c_i =
+        challenge::<C>(identifier, &R_i, &commitment.first()?.0).ok_or(Error::DKGNotSupported)?;
```

Comment:
Are the verifying_key and commitment the wrong way round here? I think it's meant to be 
```let c_i =challenge::<C>(identifier, &commitment.first()?.0, &R_i).ok_or(Error::DKGNotSupported)?;```

Reply:
Well spotted, thank you! I fixed this in 28c582d. To be sure we get it right even if refactor things, I changed the `challenge` function to take a VerifyingKey instead of an Element. Then I also changed the `first()` method to `verifying_key()` which also has the advantage of being more descriptive.

I also went ahead and did the same thing for the regular FROST `challenge()` function in b1c3989, looks much nicer.

Lesson learned, I'll try to avoid writing functions with multiple arguments of the same type in the future.

Reply:
Also here is the PR were we fixed the order, to double check we're hashing in the correct order https://github.com/ZcashFoundation/frost/pull/484/files